### PR TITLE
Fix bug where we could fail to emit user requested secrets or configmaps

### PR DESCRIPTION
**Pull request by matthchr**
If the user requests a particular secret or configmap value be written, we should error if we are unable to write it and retry.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests
- [ ] this PR contains YAML Samples

### CODE REVIEW

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/internal/controllers/edge_case_test.go [hunk 1]
```diff
@@ -277,3 +280,66 @@ func Test_Owner_IsMutableIfNotSuccessfullyCreated(t *testing.T) {
...
+	skipAcct.Name = skipAcct.Name + "-skip" // So we don't collide
+	skipAcct.Spec.OperatorSpec.Secrets.PrimaryMasterKey.Name = cosmosSecret2
+	skipAcct.Spec.OperatorSpec.Secrets.PrimaryReadonlyMasterKey.Name = cosmosSecret2
+	skipAcct.Spec.OperatorSpec.Secrets.DocumentEndpoint.Name = cosmosSecret2
+	skipAcct.Annotations = map[string]string{
+		annotations.ReconcilePolicy: string(annotations.ReconcilePolicySkip),
+	}
+
+	tc.ExportAsSampleNamed(acct, "s1")
+	tc.ExportAsSampleNamed(skipAcct, "s2")
```
[the remark this answers is not in the document]
Comment (matthchr) [reply v2/internal/controllers/edge_case_test.go:339]: No this was for testing purposes I didn't mean to leave them here.

File: v2/samples/machinelearningservices/v1api/v1api20210701_workspace.yaml [hunk 2]
```diff
@@ -24,18 +24,6 @@ spec:
   # Optional: Save the keys for the storage account into a Kubernetes secret
   operatorSpec:
     secrets:
-      appInsightsInstrumentationKey:
-        name: workspace-secret
```
Comment (super-harsh) [inline v2/samples/machinelearningservices/v1api/v1api20210701_workspace.yaml:28]: Was there a reason to remove them? 
Rather would suggest commenting them out, so users find it easy to uncomment and export.

**Pull request merged**